### DISCUSSION TITLE
Linguist Exclusion in .gitattributes Not Changing Language Statistics

### DISCUSSION THREAD

**Discussion by alexdlaird** [github-linguist/linguist#6681] [General]
I have a Python library that has some test resources in HTML. I've added an exclusion for this entire folder in `.gitattributes`: https://github.com/alexdlaird/amazon-orders-python/blob/main/%20.gitattributes

But the language statistics still show HTML as the primary language for the repo, and clicking on this only shows the files I've excluded. Does this take time to update? Am I missing a step?

<img width="379" alt="Screenshot 2024-01-17 at 8 12 07 AM" src="https://github.com/github-linguist/linguist/assets/1660326/8e3458f4-7fa2-4569-8111-60aaaf2dd38a">

**Comment by lildude** [comment]
It's hard to see in the UI and even CLI but you've got an extra space at the beginning of the `.gitattributes` filename in your repo.

You can almost see it with `ls -1a` (the dot doesn't align):

```console
➜ ls -1a | head -5
 .gitattributes
.
..
.coveragerc
.git
➜
```

... but `eza` which quotes filenames with spaces really highlights the issue:

```console
➜ eza -1a | head -3
.coveragerc
.git
' .gitattributes'
.github
.gitignore
➜
```

Correctinging the filename will fix the issue:

```console
➜ git mv ' .gitattributes' .gitattributes
➜ git commit -m 'Rename'
[main 2b0b029] Rename
 1 file changed, 0 insertions(+), 0 deletions(-)
 rename  .gitattributes => .gitattributes (100%)
➜ git check-attr --all tests/resources/new-otp-failed.html
tests/resources/new-otp-failed.html: diff: html
tests/resources/new-otp-failed.html: linguist-vendored: set
➜ github-linguist . --breakdown
97.65%  81979      Python
2.35%   1976       Makefile

Makefile:
Makefile

Python:
amazonorders/cli.py
amazonorders/entity/item.py
amazonorders/entity/order.py
amazonorders/entity/parsable.py
amazonorders/entity/recipient.py
amazonorders/entity/seller.py
amazonorders/entity/shipment.py
amazonorders/exception.py
amazonorders/orders.py
amazonorders/session.py
scripts/build-test-resources.py
setup.py
tests/test_cli.py
tests/test_integration.py
tests/test_orders.py
tests/test_session.py
tests/testcase.py
tests/unittestcase.py
➜ 
```

**Comment by alexdlaird** [comment]
Wow! Ha, thanks for finding that!
